pages/_app.js

```diff
@@ -1,10 +1,44 @@
 import { GlobalStyle } from "@/GlobalStyles";
+import { useState } from "react";
+import { pets } from "@/lib/Data";
+import { useRouter } from "next/router";
+import { uid } from "uid";
 
 export default function App({ Component, pageProps }) {
+  const [petCollection, setPetCollection] = useState(pets);
+  const router = useRouter();
+
+  function handleCreatePet(petData) {
+    setPetCollection((prevData) => [
+      {
+        id: uid(),
+        name: petData.name,
```

Comment:
I agree to P-Oliomonte: 
since you get the petData as an argument you can enrich it here like:
```js
//...
{ ...petData,
  id: uid()
}
```
Also you add the status to the petData inside the handleSubmit. At the moment in the form are different values for hunger, happiness, etc displayed but you give them all 80% here. I think that's not what you want. You shoud read the values inside the handlesubmit and add it to the petData.